aws_lambda_powertools/utilities/parameters/base.py

```diff
@@ -0,0 +1,124 @@
+    ) -> Union[str, list, dict, bytes]:
+        """
+        Retrieve a parameter value or return the cached value
+
+        Parameters
+        ----------
+
+        name: str
+            Parameter name
+        max_age: int
+            Maximum age of the cached value
+        transform: str
+            Optional transformation of the parameter value. Supported values
+            are "json" for JSON strings and "binary" for base 64 encoded
+            values.
+
+        Raises
+        ------
+
+        GetParameterError
+            When the parameter provider fails to retrieve a parameter value for
+            a given name.
+        """
+
+        # If there are multiple calls to the same parameter but in a different
+        # transform, they will be stored multiple times. This allows us to
+        # optimize by transforming the data only once per retrieval, thus there
+        # is no need to transform cached values multiple times. However, this
+        # means that we need to make multiple calls to the underlying parameter
+        # store if we need to return it in different transforms. Since the number
+        # of supported transform is small and the probability that a given
+        # parameter will always be used in a specific transform, this should be
+        # an acceptable tradeoff.
+        key = (name, transform)
+
+        if key not in self.store or self.store[key].ttl < datetime.now():
+            try:
+                value = self._get(name, **kwargs)
+            # Encapsulate all errors into a generic GetParameterError
+            except Exception as exc:
+                raise GetParameterError(str(exc))
+
+            if transform == "json":
+                value = json.loads(value)
+            elif transform == "binary":
+                value = base64.b64decode(value)
+
+            self.store[key] = ExpirableValue(value, datetime.now() + timedelta(seconds=max_age),)
+
+        return self.store[key].value
+
+    @abstractmethod
+    def _get(self, name: str, **kwargs) -> str:
+        """
+        Retrieve paramater value from the underlying parameter store
+        """
+        raise NotImplementedError()
+
+    def get_multiple(
+        self, path: str, max_age: int = DEFAULT_MAX_AGE_SECS, transform: Optional[str] = None, **kwargs
+    ) -> Union[Dict[str, str], Dict[str, dict], Dict[str, bytes]]:
+        """
+        Retrieve multiple parameters based on a path prefix
+        """
+
+        key = (path, transform)
+
+        if key not in self.store or self.store[key].ttl < datetime.now():
+            try:
+                values = self._get_multiple(path, **kwargs)
+            # Encapsulate all errors into a generic GetParameterError
+            except Exception as exc:
+                raise GetParameterError(str(exc))
+
+            if transform == "json":
+                values = {k: json.loads(v) for k, v in values.items()}
+            elif transform == "binary":
```

Comment:
Same as before - Perhaps raise an exception if string isn't `json` or `binary`? 

I was gonna suggest an enum to make this easier but it'll further complicate the code for only two options (not worth it)